src/triplestore-builtin.c

```diff
@@ -382,36 +379,38 @@ const Triple *triplestore_poll(TripleState *state)
 
 
 /*
-  This function should be called iteratively.  Before the first call
-  it should be provided a `state` initialised with triplestore_init_state().
+  Return next triple matching s-p-o triple, where `d` is the datatype
+  of the object. Any of `s`, `p`, `o` or `d` may be NULL, in which case
+  they will match anything.
+
+  If `d` starts with '@', it will match language-tagged plain text
+  literal objects whos XML language abbreviation matches the string
+  following the '@'-sign.
+
+  if `d` is NUL (empty string), it will match non-literal objects.
```

Comment:
```suggestion
  if `d` is NULL (empty string), it will match non-literal objects.
```

Reply:
This is some C jargon. NUL means an empty string (a byte of value zero) while NULL means a pointer pointing to the invalid memory address zero (eight zero bytes on a 64-bit machine).